src/app/clusters/diagnostic-logs-server/diagnostic-logs-server.h

```diff
@@ -60,6 +63,12 @@ class DiagnosticLogsServer
 
 private:
     static DiagnosticLogsServer sInstance;
+
+    void InvokeCommand(HandlerContext & ctx) override;
+
+#ifdef MATTER_DM_DIAGNOSTIC_LOGS_CLUSTER_CLIENT_ENDPOINT_COUNT
+    void HandleRetrieveLogsRequest(HandlerContext & ctx, const Commands::RetrieveLogsRequest::DecodableType & commandData);
```

Comment:
Why are we excluding just this function, and not the entire server impl?  But I would expect this cluster to not be built at all if no endpoints enable it, as things stand....

Reply:
I thought MATTER_DM_DIAGNOSTIC_LOGS_CLUSTER_CLIENT_ENDPOINT_COUNT should always be defined, but I could not explain the incurred FLASH cost from my refactor, so I just wrapped my refactored part to double confirm if this is the root cause

Reply:
Please ignore my intermediate test, I trigger another run with flag DIAGNOSTIC_LOGS_ENABLE_RETRIEVE_LOGS_REQUEST_CMD and see if there is flash increase